cmd/main_test.go

```diff
@@ -267,3 +270,83 @@ var _ = AfterSuite(func() {
 		Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error killing the server during test teardown: %v", err))
 	}
 })
+
+type mockAuthorization struct {
+	allowed bool
+	err error
+}
+
+func (m *mockAuthorization) runAccessCheck(
+	e *echo.Echo,
+	authCl authorizationv1Client.AuthorizationV1Interface,
+	email, namespace, resourceGroup, resource, verb string,
+) (bool, error) {
+	return m.allowed, m.err
+}
+
+var _ = Describe("TestGetNamespacesWithAccess", func() {
+	var (
+		e               *echo.Echo
+		c               echo.Context
+		authCl          *mockAuthorization
+		allNamespaces   []k8sapi.Namespace
+		expectedNs      []k8sapi.Namespace
+		actualNs        []k8sapi.Namespace
+		err             error
+	)
+
+	BeforeEach(func() {
```

Comment:
I'm not against using `BeforeEach` or `JustBeforeEach`, but we can also make use of `BeforeSuite` if needed. Maybe we can use the `each`  for specific setups for whole tests.